File: wastelessapp/src/app/models/item.ts
export class Item {
  ItemId: number;
  ItemName: string;
  Quantity: number;
  Calories: number;
  PurchaseDate: string;
  ExpirationDate: string;
  ConsumptionDate: string;
  Ideal: number;

  constructor(iN:string,q:number,c:number,pD:string,eD:string,cD:string,ide?:number ) {
    this.ItemName = iN;
    this.Quantity = q;
    this.Calories = c;
    this.PurchaseDate = pD;
    this.ExpirationDate = eD;
    this.ConsumptionDate = cD;
    this.Ideal = ide;
  }

  // get ideal(): number{
  //   return this._ideal;
  // }
  //
  // set ideal(val: number){
  //   this._ideal = val;
  // }

}

File: WasteLESS/WasteLESS/Models/Item.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations.Schema;

namespace WasteLESS.Models
{
    public class Item
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ItemId { get; set; }
        public string ItemName { get; set; }
        public int Quantity { get; set; }
        public int Calories { get; set; }
        public DateTime PurchaseDate { get; set; }
        public DateTime ExpirationDate { get; set; }
        public DateTime ConsumptionDate { get; set; }
        public int Ideal { get; set; }
        public IList<GroceryListItem> GroceryListItems { get; set; }

    }
}
